### ISSUE TITLE
Data pre-load with threading(? multithreading?) for large-scale data

### ISSUE THREAD

**Issue by lim0606** [Lasagne/Lasagne#313]
Hi, 

I'd like to write a code for pre-loading data for large-scale image classification model.

I wrote a code for it by modifying https://github.com/skaae/lasagne-draw/blob/master/deepmodels/batchiterator.py, but my code didn't work as I expected (my code is in here https://github.com/lim0606/lasagne-googlenet/blob/master/utils/batchiterator.py)

Is there any one who can help me? :)

Thank you,

Jaehyun

**Comment by benanne** [comment]
For support requests, please use the mailing list instead: https://groups.google.com/forum/#!forum/lasagne-users
We would like to reserve GitHub issues for development and bug reports. Thanks!

Also, please provide a little more information that "my code didn't work as I expected". We have no idea what you expect, and most of us don't have time to read almost 200 lines of code to figure out what might not work the way it should. Please be a bit more specific :)

**Comment by lim0606** [comment]
@benanne 

I apologise for the inconvenience.

The problem was that my code didn't seem to work fast enough. There were two problems; one is I tried to resize images during pre-load images, and the other is HDD problem. 

I think I solved this issue by writing a script to save imagenet images (numpy arrays to be used in Lasagne) to lmdb as well as re-designing batch iterator for lmdb. I think it basically works well, but I'm not sure 100%, since there is still a problem of performance in my computer because of hard disk problem. 

1. a script to save imagenet dataset to lmdb 
https://github.com/lim0606/lasagne-googlenet/blob/master/tools/convert_imagenet.py
2. batch iterator for lmdb 
https://github.com/lim0606/lasagne-googlenet/blob/master/utils/batchiterator.py

Jaehyun
